File: src/1903.largest-odd-number-in-string.2032/solution.js
/**
 * @param {string} num
 * @return {string}
 */
var largestOddNumber = function (num) {
    let index = num.length - 1;
    while (index > -1) {
        if ((+num[index]) & 1) {
            break;
        }
        index--;
    }

    return num.slice(0, index + 1);
};

File: src/1903.largest-odd-number-in-string.2032/solution.java
class Solution {
    public String largestOddNumber(String num) {
        char[] chars = num.toCharArray();
        int index = chars.length-1;
        while(index>0){
            if( (chars[index]-'0')%2 == 1 ){
                return num.substring(0,index+1);
            }
            index--;
        }
        return "";
    }
}